presto-main/src/main/java/com/facebook/presto/server/security/SpnegoFilter.java

```diff
@@ -211,9 +212,14 @@ public Principal getUserPrincipal()
         return Optional.empty();
     }
 
-    private static void sendChallenge(HttpServletResponse response, boolean includeRealm, String invalidSpnegoToken)
+    private static void sendChallenge(HttpServletRequest request, HttpServletResponse response, boolean includeRealm, String invalidSpnegoToken)
         throws IOException
     {
+        // skip request body before sending the status to avoid dropping of the connection
+        // if connection is dropped before all the content is sent by jetty client, the client request will be marked as failed,
+        // but not as unauthorized. And the further request with the authentication header won't be sent.
+        // https://github.com/eclipse/jetty.project/issues/938
```

Comment:
Don't link to the issue since it's long and closed as won't-fix. Any relevant info should be included here.